postgres/src/lib.rs

```diff
@@ -0,0 +1,693 @@
+                snapshot.to_vec(),
+            )
+            .await?;
+            txn.commit().await?;
+
+            txn = storage.txn(client_id).await?;
+            assert_eq!(
+                txn.get_client().await?,
+                Some(Client {
+                    latest_version_id: Uuid::nil(),
+                    snapshot: Some(Snapshot {
+                        version_id: snapshot_version_id,
+                        timestamp: Utc.timestamp_opt(snapshot_timestamp, 0).unwrap(),
+                        versions_since: versions_since_snapshot,
+                    })
+                })
+            );
+
+            let row = db_client
+                .query_one(
+                    "select snapshot from clients where client_id = $1",
+                    &[&client_id],
+                )
+                .await?;
+            assert_eq!(row.get::<_, &[u8]>(0), b"abcd");
+
+            Ok(())
+        })
+        .await
+    }
+
+    #[tokio::test]
+    async fn test_get_snapshot_none() -> anyhow::Result<()> {
+        with_db(async |connection_string, db_client| {
+            let storage = PostgresStorage::new(connection_string).await?;
+            let client_id = make_client(&db_client).await?;
+            let mut txn = storage.txn(client_id).await?;
+            assert_eq!(txn.get_snapshot_data(Uuid::new_v4()).await?, None);
+
+            Ok(())
+        })
+        .await
+    }
+
+    #[tokio::test]
+    async fn test_get_snapshot_mismatched_version() -> anyhow::Result<()> {
+        with_db(async |connection_string, db_client| {
+            let storage = PostgresStorage::new(connection_string).await?;
+            let client_id = make_client(&db_client).await?;
+            let mut txn = storage.txn(client_id).await?;
+
+            let snapshot_version_id = Uuid::new_v4();
+            let versions_since_snapshot = 10;
+            let snapshot_timestamp = 10000000;
+            let snapshot = b"abcd";
+            txn.set_snapshot(
+                Snapshot {
+                    version_id: snapshot_version_id,
+                    timestamp: Utc.timestamp_opt(snapshot_timestamp, 0).unwrap(),
+                    versions_since: versions_since_snapshot,
+                },
+                snapshot.to_vec(),
+            )
+            .await?;
+
+            assert_eq!(txn.get_snapshot_data(Uuid::new_v4()).await?, None);
+
+            Ok(())
+        })
+        .await
+    }
+
+    #[tokio::test]
+    async fn test_get_version_none() -> anyhow::Result<()> {
+        with_db(async |connection_string, db_client| {
+            let storage = PostgresStorage::new(connection_string).await?;
+            let client_id = make_client(&db_client).await?;
+            let mut txn = storage.txn(client_id).await?;
+            assert_eq!(txn.get_version_by_parent(Uuid::new_v4()).await?, None);
+            assert_eq!(txn.get_version(Uuid::new_v4()).await?, None);
+            Ok(())
+        })
+        .await
+    }
```

Comment:
These exact same assertions are made in the subsequent test (`test_get_version`).
```suggestion
```